File: InterLab-Server/InterLab.Core/Models/Stock.cs
using Newtonsoft.Json;

namespace InterLab.Core.Models
{
    public class Stock
    {
        public string? Ticker { get; set; }
        public string? Name { get; set; }
        public string? Exchange_short { get; set; }
        public string? Exchange_long { get; set; }
        public string? Mic_code { get; set; }
        public string? Currency { get; set; }
        public double Price { get; set; }
        public double Day_high { get; set; }
        public double Day_low { get; set; }
        public double Day_open { get; set; }

        [JsonProperty("52_week_high")]
        public double _52_week_high { get; set; }

        [JsonProperty("52_week_low")]
        public double _52_week_low { get; set; }
        public long? Market_cap { get; set; }
        public double Previous_close_price { get; set; }
        public DateTime Previous_close_price_time { get; set; }
        public double Day_change { get; set; }
        public int Volume { get; set; }
        public bool Is_extended_hours_price { get; set; }
        public DateTime Last_trade_time { get; set; }
        public double Price_diff => Price - Previous_close_price;
        public double Price_diff_percentage => ((Price - Previous_close_price) / Previous_close_price) * 100;
    }
}

File: InterLab-Client/src/app/models/stock.ts
export class Stock {
    ticker: string;
    name: string;
    exchange_short: string;
    exchange_long: string;
    mic_code: string;
    currency: string;
    price: number;
    day_high: number;
    day_low: number;
    day_open: number;
    _52_week_high: number;
    _52_week_low: number;
    market_cap: number;
    previous_close_price: number;
    previous_close_price_time: Date;
    day_change: number;
    volume: number;
    is_extended_hours_price: boolean;
    last_trade_time: Date;

    constructor() {
        this.ticker = '';
        this.name = '';
        this.exchange_short = '';
        this.exchange_long = '';
        this.mic_code = '';
        this.currency = '';
        this.price = 0;
        this.day_high = 0;
        this.day_low = 0;
        this.day_open = 0;
        this._52_week_high = 0;
        this._52_week_low = 0;
        this.market_cap = 0;
        this.previous_close_price = 0;
        this.previous_close_price_time = new Date();
        this.day_change = 0;
        this.volume = 0;
        this.is_extended_hours_price = false;
        this.last_trade_time = new Date();

    }
}